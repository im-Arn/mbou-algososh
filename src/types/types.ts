import { ElementStates } from "./element-states";

export type TChar = {
  char: string;
  state?: ElementStates.Changing | ElementStates.Modified | ElementStates.Default;
};

export type TState = "loading" | "load" | undefined;

export type TNumber = {
  value: number;
  state: ElementStates;
};

export type TStack = {
  elem: string;
  state?: boolean;
};

export type TQueue = {
  elem?: string;
  state?: boolean | null;
};

export interface DisableButtonState {
  addButton: boolean;
  deleteButton: boolean;
  clearButton: boolean;
}

export interface LoadersState {
  addButton: boolean;
  deleteButton: boolean;
  clearButton: boolean;
}


export type TListElement = {
  value: string;
  loading: boolean;
  load: boolean;
  upperCircle: boolean;
  bottomCircle: boolean;
};

export interface LoadersStateList {
  addHead: boolean;
  removeHead: boolean;
  addTail: boolean;
  removeTail: boolean;
  addIndex: boolean;
  removeIndex: boolean;
}

export interface DisableButtonStateList {
  addButton: boolean;
  indexButton: boolean;
  deleteButton: boolean;
}
