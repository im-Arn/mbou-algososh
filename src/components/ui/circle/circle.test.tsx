import renderer from 'react-test-renderer';
//ui
import { Circle } from './circle';
//types
import { ElementStates } from '../../../types/element-states';

describe('CIRCLE Component', () => {
    it('Рендер элемента без символов', () => {
        const tree = renderer
            .create(<Circle />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с символом', () => {
        const tree = renderer
            .create(<Circle letter='test'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с тэгом head', () => {
        const tree = renderer
            .create(<Circle head='1'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с react-элементом в head', () => {
        const tree = renderer
            .create(<Circle head={<Circle />}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с тэгом tail', () => {
        const tree = renderer
            .create(<Circle tail='1'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с react-элементом в tail', () => {
        const tree = renderer
            .create(<Circle tail={<Circle />}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с индексом', () => {
        const tree = renderer
            .create(<Circle index={1}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента с пропом isSmall ===  true', () => {
        const tree = renderer
            .create(<Circle isSmall={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента в состоянии default', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Default} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента в состоянии changing', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Changing} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Рендер элемента в состоянии modified', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Modified} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});