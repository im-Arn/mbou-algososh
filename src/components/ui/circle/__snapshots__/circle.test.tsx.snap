// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CIRCLE Component Рендер элемента без символов 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-test="index_tag"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-test="index_tag"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`CIRCLE Component Рендер элемента с индексом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с символом 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с тэгом head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`CIRCLE Component Рендер элемента с тэгом tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-test="index_tag"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    1
  </div>
</div>
`;
